File: backend/src/main/java/com/trainingplatform/entity/Course.java
package com.trainingplatform.entity;

import jakarta.persistence.CollectionTable;
import jakarta.persistence.Column;
import jakarta.persistence.ElementCollection;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OrderBy;
import jakarta.persistence.Table;
import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.List;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "courses")
public class Course extends BaseEntity {

    @Column(nullable = false)
    private String title;

    @Column(nullable = false, unique = true)
    private String slug;

    @Column(name = "short_description")
    private String shortDescription;

    @Column(columnDefinition = "TEXT")
    private String description;

    private String image;

    private String level;

    private String duration;

    private String mode;

    private BigDecimal price;

    @Column(name = "discount_price")
    private BigDecimal discountPrice;

    @Column(columnDefinition = "TEXT")
    private String requirements;

    @Column(name = "target_audience", columnDefinition = "TEXT")
    private String targetAudience;

    @Column(name = "certificate_available", nullable = false)
    private boolean certificateAvailable = false;

    @Column(name = "is_published", nullable = false)
    private boolean published = false;

    @Column(name = "instructor_name")
    private String instructorName;

    @Column(name = "instructor_bio", columnDefinition = "TEXT")
    private String instructorBio;

    @Column(name = "instructor_avatar_url")
    private String instructorAvatarUrl;

    @Column(columnDefinition = "TEXT")
    private String projects;

    @ElementCollection(fetch = FetchType.LAZY)
    @CollectionTable(name = "course_included_items", joinColumns = @JoinColumn(name = "course_id"))
    @OrderBy("position ASC")
    private List<CourseIncludedItem> whatsIncluded = new ArrayList<>();

    @Column(name = "ai_skills_description", length = 500)
    private String aiSkillsDescription;
}

File: frontend/src/types/course.ts
export interface CourseListItem {
  openCohortCount?: number;
  nextStartDate?: string | null;
  privateTutorialAvailable?: boolean;
  id: number;
  title: string;
  slug: string;
  shortDescription: string | null;
  image: string | null;
  level: string | null;
  duration: string | null;
  mode: string | null;
  price: number | null;
  discountPrice: number | null;
  certificateAvailable: boolean;
  aiSkillsDescription: string | null;
}

export interface CourseTopic {
  id: number;
  moduleId: number;
  title: string;
  position: number;
}

export interface CourseModule {
  id: number;
  courseId: number;
  title: string;
  description: string | null;
  position: number;
  topics: CourseTopic[];
}

export interface Faq {
  id: number;
  courseId: number | null;
  question: string;
  answer: string;
  position: number;
}

export interface WhatsIncludedItem {
  title: string;
  description: string | null;
}

export interface CourseDetail {
  id: number;
  title: string;
  slug: string;
  shortDescription: string | null;
  description: string | null;
  image: string | null;
  level: string | null;
  duration: string | null;
  mode: string | null;
  price: number | null;
  discountPrice: number | null;
  requirements: string | null;
  targetAudience: string | null;
  certificateAvailable: boolean;
  instructorName: string | null;
  instructorBio: string | null;
  instructorAvatarUrl: string | null;
  projects: string | null;
  whatsIncluded: WhatsIncludedItem[];
  aiSkillsDescription: string | null;
  modules: CourseModule[];
  faqs: Faq[];
}

export interface PrivateTutorialOption {
  courseId: number;
  courseTitle: string;
  courseSlug: string;
  courseImage: string | null;
  shortDescription: string | null;
  level: string | null;
  price: number | null;
  discountPrice: number | null;
  aiSkillsDescription: string | null;
  cohortId: number;
  cohortName: string;
}
